### PR TITLE
Add hasAttr and hasData

### PR DESCRIPTION
**Pull request by keichinger**
Sometimes it's useful to check whether a given attribute is just set, especially for marker attributes used by some frameworks and sites.

This PR adds support for this with `hasAttr(element, key)` and `hasData(element, key)`.

Side effect: with `hasAttr()` it's also possible to check for `data-*` attributes.

### CODE REVIEW

**Comment by apfelbox** [comment]
Do we really need this in Core?
I mean, both of them are essentially null != get*() ...

(And that’s how they should be implemented btw, instead of mostly duplicating the get*() functions)

**Comment by apfelbox** [comment]
Actually, I will probably remove getAttr in the next major version, as the function in mojave is rather useless.

**Comment by keichinger** [comment]
For a simple „is that attribute set” I'd find it easier to read and reason about the code's functionality when it's a dedicated method.

About the re-implementation: That's true, but there's IMO an important thing to notice that `element.dataset[index]` returns `undefined` instead of `null`, which makes the comparisons for `getData` and `getAttr` more ugly than a dedicated method.

**Comment by apfelbox** [comment]
And lastly, both implementations are wrong. You should use hasAttribute and probably containment checks for the object in hasData(). 

> Consequently, you should use element.hasAttribute() to check for an attribute's existence prior to calling getAttribute() if it is possible that the requested attribute does not exist on the specified element.

https://developer.mozilla.org/en-US/docs/Web/API/Element/getAttribute


I would remove hasAttr - because it’s doesn’t add anything - and change hasData to check for existence and not null (in contrast to my first comment).

**Comment by apfelbox** [comment]
> For a simple „is that attribute set” I'd find it easier to read and reason about the code's functionality when it's a dedicated method.

hasAttr(element, „test“)

is imo way less readable than

element.hasAttribute(„test“)

(Writing on an iPhone in Github is hell)

**Comment by keichinger** [comment]
Well, the existence of `element.hasAttribute` changes a lot. Don't know why I didn't think of checking for that method first.

Obviously I just wanted to write some tests... :P

I'll refactor as you said.

**Comment by apfelbox** [comment]
Souls you mind soft deprecating getAttr as well?

**Comment by keichinger** [comment]
Does that inlcude only the `@deprecated` JSDoc annotation or also an `console.warn` at the beginning or the method?

**Comment by apfelbox** [comment]
Soft deprecation is only the doc. 
Hard deprecation would be throwing warnings around. 😉

File: dom/attr.js [hunk 1]
```diff
@@ -170,6 +172,27 @@ export function getData (element, key)
...
+{
+    const value = getData(element, key);
+
+    // Empty data attributes' value is an empty string
+    if ("" === value)
+    {
+        return true;
+    }
+
+    return null !== (value || null);
```
Comment (apfelbox) [inline dom/attr.js]: `return null != value`

Comment (keichinger) [reply dom/attr.js]: Ah, that one was still in there due to the previous implementation, which could return `undefined`. Fixed :)

Comment (apfelbox) [reply dom/attr.js]: The short check `!=` is important, as it catches both `null` and `undefined`.

**Pull request merged**